test/e2e/aws_test.go

```diff
@@ -790,8 +845,15 @@ func createClusterKubeConfigs(tmpDir, namespace, clusterName string) (string, cr
 	restConfig, err := clientcmd.RESTConfigFromKubeConfig(kubeConfigData)
 	Expect(err).NotTo(HaveOccurred())
 
-	k8sclient, err := crclient.New(restConfig, crclient.Options{})
-	Expect(err).NotTo(HaveOccurred())
+	var k8sclient crclient.Client
+	Eventually(
+		func() error {
+			var err error
+			k8sclient, err = crclient.New(restConfig, crclient.Options{})
+			return err
+		},
+		5*time.Minute, 10*time.Second,
+	).Should(BeNil())
```

Comment:
Since you are testing the result of an error, I would say it is better to use `gomega.Succeed()` here.
```suggestion
	).Should(Succeed())
```